gensim/models/fasttext.py

```diff
@@ -847,20 +867,21 @@ def init_ngrams_post_load(self, file_name, wv):
         vectors are discarded here to save space.
 
         """
-        all_ngrams = []
         wv.vectors = np.zeros((len(wv.vocab), wv.vector_size), dtype=REAL)
 
         for w, vocab in wv.vocab.items():
-            all_ngrams += _compute_ngrams(w, wv.min_n, wv.max_n)
             wv.vectors[vocab.index] += np.array(wv.vectors_ngrams[vocab.index])
 
-        all_ngrams = set(all_ngrams)
-        wv.num_ngram_vectors = len(all_ngrams)
         ngram_indices = []
-        for i, ngram in enumerate(all_ngrams):
-            ngram_hash = _ft_hash(ngram)
-            ngram_indices.append(len(wv.vocab) + ngram_hash % self.bucket)
-            wv.ngrams[ngram] = i
+        wv.num_ngram_vectors = 0
+        for word in wv.vocab.keys():
+            for ngram in _compute_ngrams(word, wv.min_n, wv.max_n):
+                ngram_hash = _ft_hash(ngram) % self.bucket
+                if ngram_hash in wv.hash2index:
+                    continue
+                ngram_indices.append(len(wv.vocab) + ngram_hash)
+                wv.hash2index[ngram_hash] = wv.num_ngram_vectors
+                wv.num_ngram_vectors += 1
```

Comment:
Can be set to `len(ngram_indices)` at the end instead (sorry for nitpicking, but we already have very long code for some of these methods)